### PR TITLE
V3 drawnode prbk

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cocos/2d/CCDrawNode.cpp [hunk 1]
```diff
@@ -154,6 +178,28 @@ void DrawNode::ensureCapacity(int count)
...
 
+void DrawNode::ensureCapacityGLPoint(int count)
+{
+    CCASSERT(count>=0, "capacity must be >= 0");
+    
+    if(_bufferCountGLPoint + count > _bufferCapacityGLPoint)
+    {
+		_bufferCapacityGLPoint += MAX(_bufferCapacityGLPoint, count);
+		_bufferGLPoint = (V2F_C4B_T2F*)realloc(_bufferGLPoint, _bufferCapacityGLPoint*sizeof(V2F_C4B_T2F));
+	}
```
Comment (minggo) [inline cocos/2d/CCDrawNode.cpp]: Wrong indent.

File: cocos/2d/CCClippingNode.cpp [hunk 2]
```diff
@@ -200,12 +201,40 @@ void ClippingNode::drawFullScreenQuadClearStencil()
...
     
+    Vec2 vertices[] = {
+        Vec2(-1, -1),
+        Vec2(1, -1),
+        Vec2(1, 1),
+        Vec2(-1, 1)
+    };
     
-    DrawPrimitives::drawSolidRect(Vec2(-1,-1), Vec2(1,1), Color4F(1, 1, 1, 1));
+    GLProgram* s_shader = GLProgramCache::getInstance()->getGLProgram(GLProgram::SHADER_NAME_POSITION_U_COLOR);
```
Comment (dabingnn) [inline cocos/2d/CCClippingNode.cpp]: better to use a new name instead of `s_shader`

File: cocos/2d/CCDrawNode.cpp [hunk 3]
```diff
@@ -237,21 +339,281 @@ void DrawNode::onDraw(const Mat4 &transform, uint32_t flags)
...
+}
+
+void DrawNode::drawCircle(const Vec2 &center, float radius, float angle, unsigned int segments, bool drawLineToCenter, const Color4F &color)
+{
+    drawCircle(center, radius, angle, segments, drawLineToCenter, 1.0f, 1.0f, color);
+}
+
+void DrawNode::drawQuadBezier(const Vec2 &origin, const Vec2 &control, const Vec2 &destination, unsigned int segments, const Color4F &color)
+{
+    Vec2* vertices = new Vec2[segments + 1];
```
Comment (minggo) [inline cocos/2d/CCDrawNode.cpp]: Please use `new (no throw)`, and check return value.

File: cocos/2d/CCDrawNode.cpp [hunk 4]
```diff
@@ -454,9 +837,56 @@ void DrawNode::drawPolygon(Vec2 *verts, int count, const Color4F &fillColor, flo
...
+void DrawNode::drawSolidPoly(const Vec2 *poli, unsigned int numberOfPoints, const Color4F &color)
+{
+    drawPolygon(poli, numberOfPoints, color, 0.0, Color4F(0.0, 0.0, 0.0, 0.0));
+}
+
+void DrawNode::drawSolidCircle(const Vec2& center, float radius, float angle, unsigned int segments, float scaleX, float scaleY, const Color4F &color)
+{
+    const float coef = 2.0f * (float)M_PI/segments;
+    
+    Vec2 *vertices = (Vec2*)calloc(sizeof(Vec2)*(segments+2), 1);
```
Comment (minggo) [inline cocos/2d/CCDrawNode.cpp]: Please use new.

File: tests/cpp-tests/Classes/ClippingNodeTest/ClippingNodeTest.cpp [hunk 5]
```diff
@@ -677,15 +677,75 @@ void RawStencilBufferTest::onBeforeDrawClip(int planeIndex, const Vec2& pt)
...
     CHECK_GL_ERROR_DEBUG();
-    
-    DrawPrimitives::drawSolidRect(Vec2::ZERO, pt, _planeColor[planeIndex]);
+        
+    Vec2 vertices[] = {
+    Vec2::ZERO,
+    Vec2(pt.x, 0),
+    pt,
+    Vec2(0, pt.y)
+    };
```
Comment (minggo) [inline tests/cpp-tests/Classes/ClippingNodeTest/ClippingNodeTest.cpp:724]: Wrong indent, please add 4 spaces for initialization codes.

**Pull request merged**